Fix for playbook to work with current Mastodon

agru opened the pull request:
Several changes to get this playbook working again with the current version of Mastodon.

moritzheiber commented:
Thank you for your contribution. The tests are failing for these changes, could you take a look at it?

agru commented:
I was not aware of CircleCI and still have to understand this testing mechanism. But anyway ... I fixed the problem. The call to the playbook in CircleCI did not include the command line switch to force the usage of python3 as interpreter. I changed this in `.circleci/config.yml`. The playbook now fails at `TASK [web : Install APT repository keys]` as the testing environment can not find GPG - `Failed to find required executable gpg`. Do you have an idea how to fix this and have the test proceed at this point?

agru commented:
I now have fixed all the problems that lead to a stop of the ansible playbook. It's running fine until the end now and finishes the installation. Unfortunately there is now a problem with CircleCI that I do not know how to fix. While the ServerSpec tests are running, there is a waring `Don't run Bundler as root. Bundler can ask for sudo if it is needed ...`, it than proceeds and stops suddenly without further notice with this message: `CircleCI received exit code 1`. Any idea?

agru commented:
I see now, after refreshing CircleCI-output, that the expected ruby version there ist 2.7.2 and the expected ruby-build version is 20210405. But that's exactly something I changed in the playbook. Current version of Mastodon needs ruby 2.7.4 which can be installed with ruby-build 20210707. These version strings are changed in `bare/vars/common.yml`. I do not know how to fix this for the tests with CircleCI.

agru commented:
Ok, finally ;-) I think I now fixed everything. CircleCI was new to me, but now all checks have passed.

moritzheiber commented:
Thanks!
